docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

```diff
@@ -0,0 +1,508 @@
+| -------------- | --------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| **timestamp**  | `time`                                  | Field with timestamp, non nullable.                                                                                                                                                                                                        |
+| **body**       | `string`                                | Field with content of the log line content, non nullable.                                                                                                                                                                                  |
+| **severity**   | `string`                                | Represents the severity/level of the log line. If no severity field is found, consumers/client will decide the log level. More info about log level can be find [here](https://grafana.com/docs/grafana/latest/explore/logs-integration/). |
+| **id**         | `string`                                | Unique identifier of the log line.                                                                                                                                                                                                         |
+| **attributes** | `json raw message` (go) or `other` (ts) | This field represent additional attributes of the log line. Other systems may refer to this with other names, for example Loki calls these "labels". Value should be represented with Record<string,any> type in javascript.               |
+
+Logs data frame's `type` needs to be set to `type: DataFrameType.LogLines` in data frame's meta.
+
+Example of constructing a logs data frame in `go`:
+
+> Note: Should be used if you are creating data frames on backend in go code.
+
+```go
+frame := data.NewFrame(
+    "logs",
+    data.NewField("timestamp", nil, []time.Time{time.UnixMilli(1645030244810), time.UnixMilli(1645030247027), time.UnixMilli(1645030247027)}),
+    data.NewField("body", nil, []string{"message one", "message two", "message three"}),
+    data.NewField("severity", nil, []string{"critical", "error", "warning"}),
+    data.NewField("id", nil, []string{"xxx-001", "xyz-002", "111-003"}),
+    data.NewField("attributes", nil, []json.RawMessage{[]byte(`{}`), []byte(`{"hello":"world"}`), []byte(`{"hello":"world", "foo": 123.45, "bar" :["yellow","red"], "baz" : { "name": "alice" }}`)}),
+)
+
+	frame.SetMeta(&data.FrameMeta{
+		Type:   data.FrameTypeLogLines,
+	})
+```
+
+Example of constructing a logs data frame in `typescript`:
+
+> Note: Should be used if you are creating data frames on frontend in typescript code.
```

Comment:
I would remove or rephrase these notes, as they repeat the same we said before "Example of constructing a logs data frame in Typescript".

Maybe:

```suggestion
> Note: Should only be used if you need to create data frames in the frontend/backend part of the plugin.
```

Reply:
Same as above